crates/router_derive/src/macros/misc.rs

```diff
@@ -0,0 +1,50 @@
+pub fn get_field_type(field_type: syn::Type) -> Option<syn::Ident> {
+    if let syn::Type::Path(path) = field_type {
+        path.path.get_ident().cloned()
```

Comment:
This will fail if the type name has more than one segments (like `api_models::MerchantAccountCreate`). Can we account for such cases as well?

Reply:
I have changed the implementation to handle this. Replaced `get_ident()` with the following code
```rust
 path.path
        .segments
        .last()
        .map(|last_path_segment| last_path_segment.ident.to_owned())
```